libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -153,4 +154,35 @@ internal class MapboxDirectionsSession(
     override fun shutdown() {
         router.shutdown()
     }
+
+    /**
+     * Temporary method for handle routes with multiple waypoints.
+     * Current Directions API doesn't support more that one waypoint.
+     * Also NavNative returns "null" in "routeOptions" field.
```

Comment:
The `routeOptions` field isn’t formally part of the Directions API response format, so on iOS, we pass the original options object around the SDK code alongside the routes, independently of the navigator.